srml/support/src/storage/mod.rs

```diff
@@ -267,6 +354,27 @@ impl<K: Codec, V: Codec, U> AppendableStorageMap<K, V> for U
 	}
 }
 
+/// A storage map with a decodable length.
+pub trait DecodeLengthStorageMap<K: Codec, V: Codec>: StorageMap<K, V> {
+	/// Read the length of the value in a fast way, without decoding the entire value.
+	///
+	/// `T` is required to implement `Codec::DecodeLength`.
+	///
+	/// Has the same logic as [`StorageValue`](trait.StorageValue.html).
+	fn decode_len<KeyArg: Borrow<K>>(key: KeyArg) -> Option<usize>
+		where V: codec::DecodeLength;
+}
```

Comment:
what is the reason for it to be implemented on a where function on StorageValue but on an additional trait on StorageMap ? (maybe you just got inspiration from append)

Reply:
hmm it is probably because of linked_map which has a different internal representation. ok make sense

Reply:
I think because specialized `Map` behaviors (appendable, enumerable etc.) are already in different traits while for the `Value` we have just one and we prefer to keep it that way? tbh I saw that `append()` was added to the the main `StorageValue` trait and just did the same for `decode_len()`.

Reply:
Yes because append doesn't work for linked_map even for linked_map u32 => Vec<u32>.
because at the key is stored not only the value but also an additional structure that links elements